### PR TITLE
Feat/add simulator

### PR DESCRIPTION
**Pull request by a6-dou**
- added simulation core implementation
- abstract debugNode providers
- added nodeRFeal as an example

BREAKING CHANGES:
- Visualiztion `assetIn` and `assetOut` changed to plural naming `assetsOut` and `assetsOut` as it make more sense because the type is an `array` of `assets`

### CODE REVIEW

**Review by hewigovens — APPROVED** [review]

File: src/index.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
 import visualizer from "./visualizer";
+import Simulator from "./simulator";
 
-export { visualizer };
+export { visualizer, Simulator };
```
Comment (hewigovens) [inline src/index.ts:4]: should we also export upper case `V` for `visualizer`?

File: package.json [hunk 2]
```diff
@@ -27,6 +27,7 @@
   "dependencies": {
     "@typescript-eslint/eslint-plugin": "^5.55.0",
     "@typescript-eslint/parser": "^5.55.0",
+    "axios": "^1.3.5",
```
Comment (hewigovens) [inline package.json:30]: is it possible just use `fetch`? no 3rd party dependencies

File: package.json [hunk 3]
```diff
@@ -27,6 +27,7 @@
...
     "eslint": "^8.36.0",
```
Comment (hewigovens) [inline package.json:31]: `eslint` seems that can be moved to `devDependencies`

File: src/index.ts (hunk 1 above)
Comment (a6-dou) [reply src/index.ts:4]: visualizer is a function but Simulator is a class, will refactor this in next PR to make both classes

File: package.json (hunk 2 above)
Comment (a6-dou) [reply package.json:30]: the thing is that the sdk might be used in browser and axios have wide browser support with other features like cancel request, timeout, interceptors, XSRF protection ...etc

File: package.json (hunk 3 above)
Comment (a6-dou) [reply package.json:31]: true, will fix it

**Pull request merged**